ophyd/areadetector/base.py

```diff
@@ -134,9 +137,35 @@ def _array_shape_callback(self, **kwargs):
                        **self._metadata)
 
 
-class ADComponent(Component):
-    def __init__(self, cls, suffix=None, *, lazy=True, **kwargs):
-        super().__init__(cls, suffix=suffix, lazy=lazy, **kwargs)
+K = TypeVar("K", bound=OphydObject)
+
+
+class ADComponent(Component[K]):
+    #: Default laziness for the component.  All AreaDetector components are
+    #: by default lazy - as they contain thousands of PVs that aren't
+    #: necesssary for everyday functionality.
+    lazy_default: ClassVar[bool] = True
+
+    #: The attribute name of the component.
```

Comment:
There are not inherited from the parent?

Reply:
This is related to https://github.com/bluesky/ophyd/pull/1053#discussion_r899422182 - would be happy to try any other suggestions you might have.